src/script/standard.cpp

```diff
@@ -114,6 +114,7 @@ bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, std::vector<std::v
             vSolutionsRet.push_back(std::move(witnessprogram));
             return true;
         }
+        typeRet = TX_NONSTANDARD;
```

Comment:
IMO this could be in line 86 above where `vSolutionsRet` is also cleared/initialized. It also prevents the same problem if in the future early failure returns are added.

Reply:
Could also drop lines 154 & 155 if you move this up.

Reply:
> Could also drop lines 154 & 155 if you move this up.

That would just make it more fragile. As I understand this function signature, it is meant to return a "pair" of `typeRet` and `vSolutionsRet`. Setting that pair should happen as close to the return statement as possible (regardless of early-return or not).

Reply:
> IMO this could be in line 86 above where vSolutionsRet is also cleared/initialized. It also prevents the same problem if in the future early failure returns are added.

This will silence `valgrind` and thus *hide* the same problem in the future instead of preventing it.

Reply:
I could add a `vSolutionsRet.clear();` in this line, if people prefer...

Reply:
I just consider best practice to reset outputs in the begin if the function has multiple exits. LGTM still.